sktime/forecasting/base/adapters/_statsmodels.py

```diff
@@ -147,7 +147,9 @@ def _extract_conf_int(prediction_results, alpha) -> pd.DataFrame:
 
         raise NotImplementedError("abstract method")
 
-    def _predict_interval(self, fh, X=None, coverage=0.95):
+    # todo 0.22.0 - switch legacy_interface default to False
+    # todo 0.23.0 - remove legacy_interface arg and logic using it
+    def _predict_interval(self, fh, X=None, coverage=0.95, legacy_interface=True):
```

Comment:
Side comment, not blocking. I just realised that depending on the method, the default value of the coverage differs. It is a bit inconsistent but I am not sure if we should create an issue on that...

Reply:
is that so? Can you give two examples?

Either way, it should never reach the user, as the `predict_interval` default overrides any `_predict_interval` default.

But it still would be best to homogenize this, to avoid confusion.

Reply:
I think `stats_models` is the exception. This has the default value 0.95, however, the other forecasters adapted by this PR has 0.9 (e.g. `fbpropet`)

Reply:
FYI @yarnabrina, due to `statsmodels`

Reply:
If I remember correctly, we discussed previously on this. I kept the same default as it was in the base docstring while creating the statsmodels adapter wrapper, which differs with the function signature.

Ref. https://github.com/sktime/sktime/issues/4434

Reply:
Maybe it is safest if the private methods do *not* have a default at all? Because the public method must pass an argument to all private methods, and defaults should be global.

So, to avoid any accidents, it is perhaps best to remove all defaults from all private methods?

Reply:
I opened a design issue for this:
https://github.com/sktime/sktime/issues/4809

I would consider this out of scope for this PR, but we should deal with it. That should also resolve #4434 ultimately.